client/customloaders/customfileloader/index.js

```diff
@@ -0,0 +1,59 @@
+var Promise = require('bluebird');
+var fse = Promise.promisifyAll(require('fs-extra'));
+var loaderUtils = require('loader-utils');
+var mime = require('mime');
+var path = require('path');
+
+module.exports = function (content) {
+  this.cacheable && this.cacheable();
+  if (!this.emitFile) {
+    throw new Error('emitFile is required from module system');
+  }
+  var query = loaderUtils.parseQuery(this.query);
+  var url = loaderUtils.interpolateName(this, query.name || '[hash].[ext]', {
+    context: query.context || this.options.context,
+    content: content,
+    regExp: query.regExp
+  });
+
+  // url-loader functionality
+  var limit = (this.options && this.options.url && this.options.url.dataUrlLimit) || 0;
+  var mimetype = query.mimetype || query.minetype || mime.lookup(this.resourcePath);
```

Comment:
:o
Didn't even read that: https://github.com/webpack/url-loader/blob/master/index.js